packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -48,6 +48,48 @@ export default util.createRule({
   create(context) {
     const sourceCode = context.getSourceCode();
     return {
+      'LogicalExpression[operator=||]'(node: TSESTree.LogicalExpression): void {
```

Comment:
> The feature does not handle function calls in this way as `(foo || {}).bar()` can cause a runtime error.

Hmm, interesting point. My hunch would be that we probably want to handle function calls for cases like this:

```ts
(someContainer || {}).hasOwnProperty("someKey")
```

...but I wouldn't consider it a blocker / must-have for this PR. Up to you!

Reply:
@JoshuaKGoldberg Care to re-review this? I don't think I'm going to add object built-ins in the PR

Reply:
You got it, no problem!

Reply:
You got it, no problem!